src/components/SWActivationModal/SWActivationModal.js

```diff
@@ -18,99 +18,181 @@
 import Button from 'components/Button';
 import { MediumText, BaseText } from 'components/Typography';
 import { Spacing } from 'components/Layout';
+import FeeLabelToggle from 'components/FeeLabelToggle';
 
 // constants
 import { ASSETS } from 'constants/navigationConstants';
-
-// actions
-import { deploySmartWalletAction } from 'actions/smartWalletActions';
+import { ETH } from 'constants/assetsConstants';
+import { FEATURE_FLAGS } from 'constants/featureFlagsConstants';
 
 // utils
 import { images } from 'utils/images';
+import { buildTxFeeInfo } from 'utils/smartWallet';
+import { isEnoughBalanceForTransactionFee } from 'utils/assets';
+
+// services
+import { firebaseRemoteConfig } from 'services/firebase';
+
+// selectors
+import { accountBalancesSelector } from 'selectors/balances';
 
 // types
-import type { Accounts } from 'models/Account';
 import type { Theme } from 'models/Theme';
+import type { Balances } from 'models/Asset';
+import type { EstimatedTransactionFee } from 'models/Transaction';
+import type { Dispatch, RootReducerState } from 'reducers/rootReducer';
 
 
 type Props = {
   theme: Theme,
-  navigation: NavigationScreenProp<*>,
   isVisible: Boolean,
   onClose: () => void,
-  accounts: Accounts,
+  navigation: NavigationScreenProp<*>,
+  deploymentEstimate: ?{ raw: Object, formatted: EstimatedTransactionFee },
+  gettingDeploymentEstimate: boolean,
+  balances: Balances,
+  isOnline: boolean,
   deploySmartWallet: () => void,
-  switchAccount: (accountId: string) => void,
+  estimateSmartWalletDeployment: () => void,
 };
 
-
 const ModalContainer = styled.View`
   padding: 20px 0 40px;
 `;
 
+const Centered = styled.View`
+  align-items: center;
+`;
 
-class SWActivationModal extends React.Component<Props> {
-  activateSW = async () => {
-    const {
-      deploySmartWallet,
-      navigation,
-      onClose,
-    } = this.props;
+const SWActivationModal = ({
+  theme,
+  isVisible,
+  onClose,
+  navigation,
+  deploymentEstimate,
+  gettingDeploymentEstimate,
+  balances,
+  isOnline,
+  deploySmartWallet,
+  estimateSmartWalletDeployment,
+}: Props) => {
+  const paidByPillar = firebaseRemoteConfig.getBoolean(FEATURE_FLAGS.SMART_WALLET_ACTIVATION_PAID_BY_PILLAR);
+
+  useEffect(() => {
+    if (!paidByPillar) {
+      estimateSmartWalletDeployment();
+    }
+  }, [isVisible]);
+
+  const { smartWalletIcon } = images(theme);
+
+  const txFeeInfo = buildTxFeeInfo(deploymentEstimate?.formatted, false);
+  const isEnoughETH = paidByPillar || (
+    txFeeInfo?.fee && isEnoughBalanceForTransactionFee(balances, { txFeeInWei: txFeeInfo.fee })
+  );
+
+  const submitButtonDisabled = !isEnoughETH || !isOnline;
+  const submitButtonVisible = true;
+
+  let submitButtonTitle = t('button.activate');
+
+  if (gettingDeploymentEstimate) {
+    submitButtonTitle = t('label.gettingFee');
+  } else if (!isEnoughETH) {
+    submitButtonTitle = t('label.notEnoughToken', { token: ETH });
+  } if (!isOnline) {
```

Comment:
Is there supposed to be a new line here?

Reply:
totally missed this, thanks